# Scan sessions
Sessions are scanned to generate maps between image files and their EXIF dates. Those maps can later be used to get the closest Lapse images for a given Motion image for frame differencing or other comparison methods.

In [28]:
%load_ext autoreload
%autoreload 2

from tqdm.notebook import tqdm
import os
from datetime import datetime

from py.Dataset import Dataset
from py.DatasetStatistics import DatasetStatistics
from py.Session import Session
from py.FileUtils import list_folders, list_jpegs_recursive

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [63]:
DIR = '/home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/'

ds = Dataset(DIR)

Found 32 sessions


In [64]:
print(ds.get_sessions())

['Beaver_01', 'Marten_01', 'Raccoon_01', 'Reddeer_01', 'Roedeer_01', 'Wildboar_01', 'Badger_02', 'Beaver_02', 'Fox_02', 'Marten_02', 'Raccoon_02', 'Rat_02', 'Reddeer_02', 'Roedeer_02', 'Badger_03', 'Fox_03', 'Raccoon_03', 'Reddeer_03', 'Wildboar_03', 'Badger_04', 'Rat_04', 'Reddeer_04', 'Wildboar_04', 'Badger_05', 'Beaver_05', 'Ermine_05', 'Fox_05', 'Marten_05', 'Raccoon_05', 'Reddeer_05', 'Roedeer_05', 'Wildboar_05']


Sessions can be created from the dataset. On instancing, the session looks for previous session scans in the session_scans folder and loads them automatically if found.

In [67]:
session = ds.create_session("Marten_01")

Session 'Marten_01' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session01-VIELAAS_Marten_01
Loaded scans.


If the session has not been scanned:

In [52]:
session.scan()

Scanning motion dates...


100%|██████████| 845/845 [00:16<00:00, 51.74it/s]


Scanning lapse dates...


100%|██████████| 626/626 [00:13<00:00, 47.86it/s]

Creating lapse map...
Saving...
Saved session_scans/Rat_02/lapse_dates.pickle
Saved session_scans/Rat_02/motion_dates.pickle
Saved session_scans/Rat_02/lapse_map.pickle


## Duplicates and inconsistent duplicates
The following functions checks for consistent and inconsistent (deviant) duplicates in the session. Duplicates are two or more images with the same EXIF date. Duplicates are inconsistent when they show different scenes (estimated by comparing the JPEG file size) despite being taken at the same time (and, supposedly, the same place).

In [53]:
total, total_duplicates, _, deviants = session.check_lapse_duplicates()

100%|██████████| 578/578 [00:00<00:00, 40275.58it/s]

* 578 lapse dates
* 48 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []


Open and compare the duplicate images for the same date.

In [68]:
session.open_images_for_date(datetime(2021, 6, 22, 9, 0))

#1 /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session01-VIELAAS_Marten_01/Lapse/2021_06_22_09h_5050.jpg
#2 /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session01-VIELAAS_Marten_01/Lapse/2021_06_22_09h_5348.jpg


### Stats

| Session | Total lapse dates | Duplicates | Multiples | Inconsistent Duplicates |
| --- | --- | --- | --- | --- |
| Beaver_01 | 1734 | 0 | 0 | 0 |
| Marten_01 | 1733 | 729 | 0 | 108 |
| Raccoon_01 | 850 | 0 | 0 | 0 |
| Reddeer_01 | 1628 | 0 | 0 | 0 |
| Roedeer_01 | 1353 | 27 | 0 | 0 |
| Wildboar_01 | 1732 | 0 | 0 | 0 |
| Badger_02 | 1728 | 0 | 0 | 0 |
| Beaver_02 | 1727 | 0 | 0 | 0 |
| Fox_02 | 957 | 0 | 0 | 0 |
| Marten_02 | 1726 | 0 | 0 | 0 |
| Raccoon_02 | 1639 | 0 | 0 | 0 |
| Rat_02 | 578 | 48 | 0 | 0 |
| Reddeer_02 | 1635 | 0 | 0 | 0 |
| Roedeer_02 | 888 | 0 | 0 | 0 |
| Badger_03 | 46 | 0 | 0 | 0 |
| Fox_03 | 34 | 1 | 1 | 1 |
| Raccoon_03 | 35 | 0 | 0 | 0 |
| Reddeer_03 | 1027 | 0 | 0 | 0 |
| Wildboar_03 | 46 | 0 | 0 | 0 |
| Badger_04 | 56 | 0 | 0 | 0 |
| Rat_04 | 39 | 0 | 0 | 0 |
| Reddeer_04 | 46 | 0 | 0 | 0 |
| Wildboar_04 | 39 | 0 | 0 | 0 |
| Badger_05 | 1105 | 69 | 0 | 1 |
| Beaver_05 | 1105 | 216 | 0 | 0 |
| Ermine_05 | 867 | 0 | 0 | 0 |
| Fox_05 | 1083 | 0 | 0 | 0 |
| Marten_05 | 888 | 2 | 0 | 2 |
| Raccoon_05 | 1088 | 1 | 0 | 1 |
| Reddeer_05 | 1078 | 0 | 0 | 0 |
| Roedeer_05 | 1090 | 0 | 0 | 0 |
| Wildboar_05 | 902 | 1 | 0 | 1 |


In [60]:
# code for generating the markup for the table above

out_file = None

for sess_name in ds.get_sessions():
    session = ds.create_session(sess_name)
    if not session.scanned:
        session.scan()
    total, total_duplicates, total_multiples, deviants = session.check_lapse_duplicates()
    out = f"| {sess_name} | {total} | {total_duplicates} | {total_multiples} | {len(deviants)} |"
    print(out)
    if out_file is not None:
        with open(out_file, "a") as f:
            f.write(f"{out}\n")

Session 'Beaver_01' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session01-VIELAAS_Beaver_01
Loaded scans.


100%|██████████| 1734/1734 [00:00<00:00, 792430.06it/s]


* 1734 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Beaver_01 | 1734 | 0 | 0 | 0 |
Session 'Marten_01' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session01-VIELAAS_Marten_01
Loaded scans.


100%|██████████| 1733/1733 [00:00<00:00, 3852.62it/s]


* 1733 lapse dates
* 729 duplicates
* 0 multiples (more than two files per date)
* 108 deviant duplicates: [datetime.datetime(2021, 6, 17, 23, 0), datetime.datetime(2021, 6, 18, 0, 0), datetime.datetime(2021, 6, 18, 1, 0), datetime.datetime(2021, 6, 18, 2, 0), datetime.datetime(2021, 6, 18, 3, 0), datetime.datetime(2021, 6, 18, 4, 0), datetime.datetime(2021, 6, 18, 5, 0), datetime.datetime(2021, 6, 18, 6, 0), datetime.datetime(2021, 6, 18, 7, 0), datetime.datetime(2021, 6, 18, 8, 0), datetime.datetime(2021, 6, 18, 9, 0), datetime.datetime(2021, 6, 18, 10, 0), datetime.datetime(2021, 6, 18, 11, 0), datetime.datetime(2021, 6, 18, 12, 0), datetime.datetime(2021, 6, 18, 13, 0), datetime.datetime(2021, 6, 18, 14, 0), datetime.datetime(2021, 6, 18, 15, 0), datetime.datetime(2021, 6, 18, 16, 0), datetime.datetime(2021, 6, 18, 17, 0), datetime.datetime(2021, 6, 18, 18, 0), datetime.datetime(2021, 6, 18, 19, 0), datetime.datetime(2021, 6, 18, 20, 0), datetime.datetime(2021, 6, 18, 21, 0), datet

100%|██████████| 850/850 [00:00<00:00, 1510020.50it/s]


* 850 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Raccoon_01 | 850 | 0 | 0 | 0 |
Session 'Reddeer_01' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session01-VIELAAS_Reddeer_01
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 9380/9380 [04:08<00:00, 37.78it/s]


Scanning lapse dates...


100%|██████████| 1628/1628 [00:37<00:00, 43.07it/s]


Creating lapse map...
Saving...
Saved session_scans/Reddeer_01/lapse_dates.pickle
Saved session_scans/Reddeer_01/motion_dates.pickle
Saved session_scans/Reddeer_01/lapse_map.pickle


100%|██████████| 1628/1628 [00:00<00:00, 1938764.03it/s]


* 1628 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Reddeer_01 | 1628 | 0 | 0 | 0 |
Session 'Roedeer_01' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session01-VIELAAS_Roedeer_01
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 38820/38820 [18:29<00:00, 35.00it/s]


Scanning lapse dates...


100%|██████████| 1380/1380 [00:33<00:00, 41.40it/s]


Creating lapse map...
Saving...
Saved session_scans/Roedeer_01/lapse_dates.pickle
Saved session_scans/Roedeer_01/motion_dates.pickle
Saved session_scans/Roedeer_01/lapse_map.pickle


100%|██████████| 1353/1353 [00:00<00:00, 72058.48it/s]


* 1353 lapse dates
* 27 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Roedeer_01 | 1353 | 27 | 0 | 0 |
Session 'Wildboar_01' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session01-VIELAAS_Wildboar_01
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 2895/2895 [01:25<00:00, 34.00it/s]


Scanning lapse dates...


100%|██████████| 1732/1732 [00:41<00:00, 41.44it/s]


Creating lapse map...
Saving...
Saved session_scans/Wildboar_01/lapse_dates.pickle
Saved session_scans/Wildboar_01/motion_dates.pickle
Saved session_scans/Wildboar_01/lapse_map.pickle


100%|██████████| 1732/1732 [00:00<00:00, 603316.55it/s]


* 1732 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Wildboar_01 | 1732 | 0 | 0 | 0 |
Session 'Badger_02' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session02-VIELAAS_Badger_02
Loaded scans.


100%|██████████| 1728/1728 [00:00<00:00, 689343.48it/s]


* 1728 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Badger_02 | 1728 | 0 | 0 | 0 |
Session 'Beaver_02' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session02-VIELAAS_Beaver_02
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 2890/2890 [01:16<00:00, 37.72it/s]


Scanning lapse dates...


100%|██████████| 1727/1727 [00:43<00:00, 39.28it/s]


Creating lapse map...
Saving...
Saved session_scans/Beaver_02/lapse_dates.pickle
Saved session_scans/Beaver_02/motion_dates.pickle
Saved session_scans/Beaver_02/lapse_map.pickle


100%|██████████| 1727/1727 [00:00<00:00, 733228.36it/s]


* 1727 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Beaver_02 | 1727 | 0 | 0 | 0 |
Session 'Fox_02' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session02-VIELAAS_Fox_02
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 1110/1110 [00:28<00:00, 38.46it/s]


Scanning lapse dates...


100%|██████████| 957/957 [00:27<00:00, 35.11it/s]


Creating lapse map...
Saving...
Saved session_scans/Fox_02/lapse_dates.pickle
Saved session_scans/Fox_02/motion_dates.pickle
Saved session_scans/Fox_02/lapse_map.pickle


100%|██████████| 957/957 [00:00<00:00, 134642.05it/s]


* 957 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Fox_02 | 957 | 0 | 0 | 0 |
Session 'Marten_02' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session02-VIELAAS_Marten_02
Loaded scans.


100%|██████████| 1726/1726 [00:00<00:00, 1315292.28it/s]


* 1726 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Marten_02 | 1726 | 0 | 0 | 0 |
Session 'Raccoon_02' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session02-VIELAAS_Raccoon_02
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 332/332 [00:06<00:00, 49.31it/s]


Scanning lapse dates...


100%|██████████| 1639/1639 [00:35<00:00, 46.32it/s]


Creating lapse map...
Saving...
Saved session_scans/Raccoon_02/lapse_dates.pickle
Saved session_scans/Raccoon_02/motion_dates.pickle
Saved session_scans/Raccoon_02/lapse_map.pickle


100%|██████████| 1639/1639 [00:00<00:00, 1118344.60it/s]


* 1639 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Raccoon_02 | 1639 | 0 | 0 | 0 |
Session 'Rat_02' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session02-VIELAAS_Rat_02
Loaded scans.


100%|██████████| 578/578 [00:00<00:00, 11164.62it/s]


* 578 lapse dates
* 48 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Rat_02 | 578 | 48 | 0 | 0 |
Session 'Reddeer_02' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session02-VIELAAS_Reddeer_02
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 675/675 [00:17<00:00, 39.35it/s]


Scanning lapse dates...


100%|██████████| 1635/1635 [00:38<00:00, 42.67it/s]


Creating lapse map...
Saving...
Saved session_scans/Reddeer_02/lapse_dates.pickle
Saved session_scans/Reddeer_02/motion_dates.pickle
Saved session_scans/Reddeer_02/lapse_map.pickle


100%|██████████| 1635/1635 [00:00<00:00, 1200365.31it/s]


* 1635 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Reddeer_02 | 1635 | 0 | 0 | 0 |
Session 'Roedeer_02' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session02-VIELAAS_Roedeer_02
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 2770/2770 [00:51<00:00, 53.39it/s]


Scanning lapse dates...


100%|██████████| 888/888 [00:20<00:00, 42.57it/s]


Creating lapse map...
Saving...
Saved session_scans/Roedeer_02/lapse_dates.pickle
Saved session_scans/Roedeer_02/motion_dates.pickle
Saved session_scans/Roedeer_02/lapse_map.pickle


100%|██████████| 888/888 [00:00<00:00, 892533.42it/s]


* 888 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Roedeer_02 | 888 | 0 | 0 | 0 |
Session 'Badger_03' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session03-VIELAAS_Badger_03
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 4245/4245 [01:02<00:00, 67.84it/s]


Scanning lapse dates...


100%|██████████| 46/46 [00:00<00:00, 146.82it/s]


Creating lapse map...
Saving...
Saved session_scans/Badger_03/lapse_dates.pickle
Saved session_scans/Badger_03/motion_dates.pickle
Saved session_scans/Badger_03/lapse_map.pickle


100%|██████████| 46/46 [00:00<00:00, 299593.14it/s]


* 46 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Badger_03 | 46 | 0 | 0 | 0 |
Session 'Fox_03' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session03-VIELAAS_Fox_03
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 5495/5495 [01:20<00:00, 68.18it/s]


Scanning lapse dates...


100%|██████████| 38/38 [00:00<00:00, 130.40it/s]


Creating lapse map...
Saving...
Saved session_scans/Fox_03/lapse_dates.pickle
Saved session_scans/Fox_03/motion_dates.pickle
Saved session_scans/Fox_03/lapse_map.pickle


100%|██████████| 34/34 [00:00<00:00, 37876.85it/s]


* 34 lapse dates
* 1 duplicates
* 1 multiples (more than two files per date)
* 1 deviant duplicates: [datetime.datetime(2021, 6, 1, 10, 0)]
| Fox_03 | 34 | 1 | 1 | 1 |
Session 'Raccoon_03' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session03-VIELAAS_Raccoon_03
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 475/475 [00:05<00:00, 79.66it/s]


Scanning lapse dates...


100%|██████████| 35/35 [00:00<00:00, 123.96it/s]


Creating lapse map...
Saving...
Saved session_scans/Raccoon_03/lapse_dates.pickle
Saved session_scans/Raccoon_03/motion_dates.pickle
Saved session_scans/Raccoon_03/lapse_map.pickle


100%|██████████| 35/35 [00:00<00:00, 133942.19it/s]


* 35 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Raccoon_03 | 35 | 0 | 0 | 0 |
Session 'Reddeer_03' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session03-VIELAAS_Reddeer_03
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 1770/1770 [00:50<00:00, 35.24it/s]


Scanning lapse dates...


100%|██████████| 1027/1027 [00:22<00:00, 45.08it/s]


Creating lapse map...
Saving...
Saved session_scans/Reddeer_03/lapse_dates.pickle
Saved session_scans/Reddeer_03/motion_dates.pickle
Saved session_scans/Reddeer_03/lapse_map.pickle


100%|██████████| 1027/1027 [00:00<00:00, 1474178.72it/s]


* 1027 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Reddeer_03 | 1027 | 0 | 0 | 0 |
Session 'Wildboar_03' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session03-VIELAAS_Wildboar_03
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 515/515 [00:06<00:00, 77.99it/s]


Scanning lapse dates...


100%|██████████| 46/46 [00:00<00:00, 148.00it/s]


Creating lapse map...
Saving...
Saved session_scans/Wildboar_03/lapse_dates.pickle
Saved session_scans/Wildboar_03/motion_dates.pickle
Saved session_scans/Wildboar_03/lapse_map.pickle


100%|██████████| 46/46 [00:00<00:00, 354014.65it/s]


* 46 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Wildboar_03 | 46 | 0 | 0 | 0 |
Session 'Badger_04' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session04-VIELAAS_Badger_04
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 480/480 [00:06<00:00, 78.13it/s]


Scanning lapse dates...


100%|██████████| 56/56 [00:00<00:00, 148.72it/s]


Creating lapse map...
Saving...
Saved session_scans/Badger_04/lapse_dates.pickle
Saved session_scans/Badger_04/motion_dates.pickle
Saved session_scans/Badger_04/lapse_map.pickle


100%|██████████| 56/56 [00:00<00:00, 302681.73it/s]


* 56 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Badger_04 | 56 | 0 | 0 | 0 |
Session 'Rat_04' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session04-VIELAAS_Rat_04
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 640/640 [00:08<00:00, 77.00it/s]


Scanning lapse dates...


100%|██████████| 39/39 [00:00<00:00, 161.61it/s]


Creating lapse map...
Saving...
Saved session_scans/Rat_04/lapse_dates.pickle
Saved session_scans/Rat_04/motion_dates.pickle
Saved session_scans/Rat_04/lapse_map.pickle


100%|██████████| 39/39 [00:00<00:00, 351025.44it/s]


* 39 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Rat_04 | 39 | 0 | 0 | 0 |
Session 'Reddeer_04' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session04-VIELAAS_Reddeer_04
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 1690/1690 [00:23<00:00, 71.78it/s]


Scanning lapse dates...


100%|██████████| 46/46 [00:00<00:00, 159.03it/s]


Creating lapse map...
Saving...
Saved session_scans/Reddeer_04/lapse_dates.pickle
Saved session_scans/Reddeer_04/motion_dates.pickle
Saved session_scans/Reddeer_04/lapse_map.pickle


100%|██████████| 46/46 [00:00<00:00, 363348.37it/s]


* 46 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Reddeer_04 | 46 | 0 | 0 | 0 |
Session 'Wildboar_04' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session04-VIELAAS_Wildboar_04
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 2250/2250 [00:31<00:00, 71.76it/s]


Scanning lapse dates...


100%|██████████| 39/39 [00:00<00:00, 137.72it/s]


Creating lapse map...
Saving...
Saved session_scans/Wildboar_04/lapse_dates.pickle
Saved session_scans/Wildboar_04/motion_dates.pickle
Saved session_scans/Wildboar_04/lapse_map.pickle


100%|██████████| 39/39 [00:00<00:00, 206277.25it/s]


* 39 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Wildboar_04 | 39 | 0 | 0 | 0 |
Session 'Badger_05' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session05-VIELAAS_Badger_05
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 3860/3860 [02:03<00:00, 31.33it/s]


Scanning lapse dates...


100%|██████████| 1174/1174 [00:27<00:00, 42.37it/s]


Creating lapse map...
Saving...
Saved session_scans/Badger_05/lapse_dates.pickle
Saved session_scans/Badger_05/motion_dates.pickle
Saved session_scans/Badger_05/lapse_map.pickle


100%|██████████| 1105/1105 [00:00<00:00, 34186.81it/s]


* 1105 lapse dates
* 69 duplicates
* 0 multiples (more than two files per date)
* 1 deviant duplicates: [datetime.datetime(2021, 6, 1, 14, 0)]
| Badger_05 | 1105 | 69 | 0 | 1 |
Session 'Beaver_05' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session05-VIELAAS_Beaver_05
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 2415/2415 [01:12<00:00, 33.33it/s]


Scanning lapse dates...


100%|██████████| 1321/1321 [00:33<00:00, 39.14it/s]


Creating lapse map...
Saving...
Saved session_scans/Beaver_05/lapse_dates.pickle
Saved session_scans/Beaver_05/motion_dates.pickle
Saved session_scans/Beaver_05/lapse_map.pickle


100%|██████████| 1105/1105 [00:00<00:00, 8703.20it/s]


* 1105 lapse dates
* 216 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Beaver_05 | 1105 | 216 | 0 | 0 |
Session 'Ermine_05' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session05-VIELAAS_Ermine_05
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 2380/2380 [01:08<00:00, 34.56it/s]


Scanning lapse dates...


100%|██████████| 867/867 [00:20<00:00, 43.27it/s]


Creating lapse map...
Saving...
Saved session_scans/Ermine_05/lapse_dates.pickle
Saved session_scans/Ermine_05/motion_dates.pickle
Saved session_scans/Ermine_05/lapse_map.pickle


100%|██████████| 867/867 [00:00<00:00, 946502.23it/s]


* 867 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Ermine_05 | 867 | 0 | 0 | 0 |
Session 'Fox_05' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session05-VIELAAS_Fox_05
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 753/753 [00:22<00:00, 34.10it/s]


Scanning lapse dates...


100%|██████████| 1083/1083 [00:25<00:00, 41.94it/s]


Creating lapse map...
Saving...
Saved session_scans/Fox_05/lapse_dates.pickle
Saved session_scans/Fox_05/motion_dates.pickle
Saved session_scans/Fox_05/lapse_map.pickle


100%|██████████| 1083/1083 [00:00<00:00, 1333655.68it/s]


* 1083 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Fox_05 | 1083 | 0 | 0 | 0 |
Session 'Marten_05' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session05-VIELAAS_Marten_05
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 16170/16170 [07:54<00:00, 34.10it/s]


Scanning lapse dates...


100%|██████████| 890/890 [00:20<00:00, 43.81it/s]


Creating lapse map...
Saving...
Saved session_scans/Marten_05/lapse_dates.pickle
Saved session_scans/Marten_05/motion_dates.pickle
Saved session_scans/Marten_05/lapse_map.pickle


100%|██████████| 888/888 [00:00<00:00, 144127.47it/s]


* 888 lapse dates
* 2 duplicates
* 0 multiples (more than two files per date)
* 2 deviant duplicates: [datetime.datetime(2021, 6, 1, 12, 0), datetime.datetime(2021, 6, 13, 4, 0)]
| Marten_05 | 888 | 2 | 0 | 2 |
Session 'Raccoon_05' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session05-VIELAAS_Raccoon_05
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 1600/1600 [00:45<00:00, 34.82it/s]


Scanning lapse dates...


100%|██████████| 1089/1089 [00:24<00:00, 44.23it/s]


Creating lapse map...
Saving...
Saved session_scans/Raccoon_05/lapse_dates.pickle
Saved session_scans/Raccoon_05/motion_dates.pickle
Saved session_scans/Raccoon_05/lapse_map.pickle


100%|██████████| 1088/1088 [00:00<00:00, 683144.12it/s]


* 1088 lapse dates
* 1 duplicates
* 0 multiples (more than two files per date)
* 1 deviant duplicates: [datetime.datetime(2021, 6, 5, 14, 0)]
| Raccoon_05 | 1088 | 1 | 0 | 1 |
Session 'Reddeer_05' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session05-VIELAAS_Reddeer_05
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 18251/18251 [07:50<00:00, 38.81it/s]


Scanning lapse dates...


100%|██████████| 1078/1078 [00:24<00:00, 43.29it/s]


Creating lapse map...
Saving...
Saved session_scans/Reddeer_05/lapse_dates.pickle
Saved session_scans/Reddeer_05/motion_dates.pickle
Saved session_scans/Reddeer_05/lapse_map.pickle


100%|██████████| 1078/1078 [00:00<00:00, 2044984.04it/s]


* 1078 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Reddeer_05 | 1078 | 0 | 0 | 0 |
Session 'Roedeer_05' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session05-VIELAAS_Roedeer_05
Loaded scans.


100%|██████████| 1090/1090 [00:00<00:00, 1023000.98it/s]


* 1090 lapse dates
* 0 duplicates
* 0 multiples (more than two files per date)
* 0 deviant duplicates: []
| Roedeer_05 | 1090 | 0 | 0 | 0 |
Session 'Wildboar_05' at folder: /home/AMMOD_data/camera_traps/BayerWald/Vielkadaver-Projekt/VIELAAS_Spring_Session05-VIELAAS_Wildboar_05
Session not scanned. Run session.scan() to create scan files
Scanning motion dates...


100%|██████████| 22263/22263 [10:24<00:00, 35.63it/s]


Scanning lapse dates...


100%|██████████| 903/903 [00:21<00:00, 42.06it/s]


Creating lapse map...
Saving...
Saved session_scans/Wildboar_05/lapse_dates.pickle
Saved session_scans/Wildboar_05/motion_dates.pickle
Saved session_scans/Wildboar_05/lapse_map.pickle


100%|██████████| 902/902 [00:00<00:00, 442642.12it/s]

* 902 lapse dates
* 1 duplicates
* 0 multiples (more than two files per date)
* 1 deviant duplicates: [datetime.datetime(2021, 6, 13, 15, 0)]
| Wildboar_05 | 902 | 1 | 0 | 1 |
